Code/Framework/AzToolsFramework/AzToolsFramework/ComponentModes/BaseViewportEdit.h

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+namespace AzToolsFramework
+{
+    //! Base class for objects used in sub component modes which use viewport editing.
+    class BaseViewportEdit
+    {
+    public:
+        virtual ~BaseViewportEdit() = default;
+
+        virtual void Setup(const AZ::EntityComponentIdPair& entityComponentIdPair) = 0;
```

Comment:
question: Is this something we might want to change where `Setup` doesn't actually take an `EntityComponentIdPair` but instead takes nothing and we have a separate function `AddEntityComponentIdPair` that can be called instead (to save people passing through an invalid one if it isn't needed?) Maybe a follow-up change, just a thought based on earlier conversations.

Reply:
Yeah, that's a nice way of doing it actually. I'll do that when I move on to the capsule and sphere component modes and do the other restructuring we discussed yesterday.

Reply:
Okay nice one good plan 👍